mod_form.php

```diff
@@ -719,7 +719,19 @@ public function validation($data, $files) {
             if ($data['start_time'] < strtotime('today')) {
                 $errors['start_time'] = get_string('err_start_time_past', 'zoom');
             }
-
+            // Make sure duration is positive and no more than 150 hours.
+            if ($data['duration'] <= 0) {
+                $errors['duration'] = get_string('err_duration_nonpositive', 'zoom');
+            } else if ($data['duration'] > 150 * 60 * 60) {
+                $errors['duration'] = get_string('err_duration_too_long', 'zoom');
+            }
+        }
+        else if($data['recurring'] == 1 && $data['recurrence_type'] != ZOOM_RECURRINGTYPE_NOTIME)
+        {
+            // Make sure start date time (first potential date of next meeting) is in the future.
+            if ($data['start_time'] < strtotime('today')) {
+                $errors['start_time'] = get_string('err_start_time_past_recurring', 'zoom');
+            }
```

Comment:
The automatic checks failed because of the else if formatting. It needs to be "cuddled", but the "if" doesn't touch the parenthesis. Also, maintain the blank lines between separate logical conditionals.
```suggestion

            // Make sure duration is positive and no more than 150 hours.
            if ($data['duration'] <= 0) {
                $errors['duration'] = get_string('err_duration_nonpositive', 'zoom');
            } else if ($data['duration'] > 150 * 60 * 60) {
                $errors['duration'] = get_string('err_duration_too_long', 'zoom');
            }
        } else if ($data['recurring'] == 1 && $data['recurrence_type'] != ZOOM_RECURRINGTYPE_NOTIME) {
            // Make sure start date time (first potential date of next meeting) is in the future.
            if ($data['start_time'] < strtotime('today')) {
                $errors['start_time'] = get_string('err_start_time_past_recurring', 'zoom');
            }

```